src/main.rs

```diff
@@ -104,6 +107,13 @@ fn run_command(cli: &Cli, log: impl Logger + Sync + Clone) -> ExitDirectives {
     let cmd_result = match &cli.command {
         Some(Commands::Registry(params)) => semconv_registry(log.clone(), params),
         Some(Commands::Diagnostic(params)) => diagnostic::diagnostic(log.clone(), params),
+        Some(Commands::Completion(completions)) => {
+            generate_completion(&completions.shell);
+            return ExitDirectives {
+                exit_code: 0,
+                quiet_mode: cli.quiet,
```

Comment:
```suggestion
                quiet_mode: true,
```
Quiet mode needs to be true to avoid weaver writing `Total execution time: 0.010793s` to stdout and spoiling the completion script.